include/nudb/basic_store.hpp

```diff
@@ -251,37 +260,47 @@ class basic_store
         The buffer provided to the callback remains valid
         until the callback returns, ownership is not transferred.
 
-        @param ec Set to the error, if any occurred. If an error
-        occurs, the return value is undefined.
-
-        @return `true` if a matching key was found.
+        @param ec Set to the error, if any occurred.
 
         @throws `std::logic_error` if the database is not open.
```

Comment:
So fetches must catch exceptions and respect the error code param. Why not just set the ec code here instead and make use pleasant for the user?